prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -321,61 +326,79 @@ class DetailedSubmissionHistory(
         }
+        // Grab the filter logs generated during the "route" action, as well as errors and warnings
+        val filterLogs = descendant.logs.filter { log -> log.type == ActionLogLevel.filter }
+        errors += descendant.errors
+        warnings += descendant.warnings
 
-        // if all the items were filtered out, no translate actions are generated (thus no destinations)
-        // to return the expected output, we create dummy destinations that are only used in enrichment
-        // these dummies are created using the receivers defined in the logs
-        if (destinations.isEmpty() && filterLogs.isNotEmpty()) {
-            val filteredReportRows = filterLogs.map { log -> log.detail.message }
-            val filteredReportItems = filterLogs.map { log ->
-                ReportStreamFilterResultForResponse(log.detail as ReportStreamFilterResult)
-            }
+        // add filter logs to its respective destination otherwise add new destination
+        if (filterLogs.isNotEmpty()) {
             filterLogs.forEach { log ->
-                val filterResult = log.detail as ReportStreamFilterResult
+                check(log.detail is ReportStreamFilterResult) { "Filter result not of type ReportStreamFilterResult" }
+                val filterResult = log.detail
+                val filterReport = log.detail.message
                 val receiverNameSegments = filterResult.receiverName.split(Sender.fullNameSeparator)
-                destinations.add(
-                    Destination(
-                        receiverNameSegments[0],
-                        receiverNameSegments[1],
-                        filteredReportRows,
-                        filteredReportItems,
-                        null,
-                        0,
-                        filterResult.originalCount,
+                val filterResultResponse = ReportStreamFilterResultForResponse(filterResult)
+
+                destinations.firstOrNull {
+                    it.organizationId == receiverNameSegments[0] && it.service == receiverNameSegments[1]
+                }?.let { existingDestination ->
+                    if (existingDestination.filteredReportRows == null) {
+                        existingDestination.filteredReportRows = mutableListOf(filterReport)
+                    } else {
+                        existingDestination.filteredReportRows!!.add(filterReport)
+                    }
+
+                    if (existingDestination.filteredReportItems == null) {
+                        existingDestination.filteredReportItems = mutableListOf(filterResultResponse)
+                    } else {
+                        existingDestination.filteredReportItems!!.add(filterResultResponse)
+                    }
+
+                    existingDestination.itemCountBeforeQualFilter = existingDestination.itemCountBeforeQualFilter?.plus(
+                        filterResult.originalCount
                     )
```

Comment:
In my other comment, what I was getting at is that if `existingDestination.itemCountBeforeQualFilter` is null, then `existingDestination.itemCountBeforeQualFilter?.plus(anything)` is going to be null as well. I think the change below would increment it treating a starting point of null as 0. It might be that we don't expect it to ever be null, if so then this isn't necessary.
```suggestion
                    existingDestination.itemCountBeforeQualFilter = (existingDestination.itemCountBeforeQualFilter ?: 0).plus(
                        filterResult.originalCount
                    )
```

Reply:
I see that now. It would so much easier if these weren't nullables. I added an extra `?:` to set it to `filterResult.originalCount` if null. I did the same for the other case.